homeassistant/components/ituran/entity.py

```diff
@@ -0,0 +1,49 @@
+"""Base for all turan entities."""
+
+from __future__ import annotations
+
+from pyituran import Vehicle
+
+from homeassistant.core import callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import IturanDataUpdateCoordinator
+
+
+class IturanBaseEntity(CoordinatorEntity[IturanDataUpdateCoordinator]):
+    """Common base for Ituran entities."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: IturanDataUpdateCoordinator,
+        vehicle: Vehicle,
+    ) -> None:
+        """Initialize entity."""
+        super().__init__(coordinator)
+
+        self.vehicle = vehicle
+
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, vehicle.license_plate)},
+            manufacturer=vehicle.make,
+            model=vehicle.model,
+            name=vehicle.model,
+            serial_number=vehicle.license_plate,
+        )
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self.vehicle = self.coordinator.data.get(
+            self.vehicle.license_plate, self.vehicle
+        )
+        super()._handle_coordinator_update()
```

Comment:
Why don't you make vehicle a property and just fetch it directly from the coordinator when needed? You should also implement `available` to return False when the license plate is not in the coordinator data

Reply:
Done (assuming I understood correctly what you meant)